### PR TITLE
Integ match bool prefix #187

### PR DESCRIPTION
**Pull request by forestmvey**
### Description

1. Support `match_bool_prefix` in sql.
2. Support `match_bool_prefix` in ppl.

 
### Issues Resolved
Resolves: [187](https://github.com/opensearch-project/sql/issues/187)
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dai-chen — DISMISSED** [review]
Just a minor comment. Thanks for the changes!

File: docs/user/dql/functions.rst [hunk 1]
```diff
@@ -2235,3 +2235,42 @@ Another example to show how to set custom values for the optional parameters::
...
+
+MATCH_BOOL_PREFIX
+-----
+
+Description
+>>>>>>>>>>>
+
+``match_bool_prefix(field_expression, query_expression)``
+
+The match_bool_prefix function maps to the match_bool_prefix query in the search engine. match_bool_prefix creates a match query from all but the last term in the query string. The last term is used to create a prefix query.
```
[the remark this answers is not in the document]
Comment (forestmvey) [reply docs/user/dql/functions.rst:2247]: > Do you know if the meaning of these params documented in OpenSearch doc or not? I don't see it here: https://opensearch.org/docs/latest/opensearch/query-dsl/full-text/#match-boolean-prefix. I'm thinking if it's missing, probably we can add it in our doc.

@dai-chen
Optional arguments are available in the options section of that page.

https://opensearch.org/docs/latest/opensearch/query-dsl/full-text/#options

**Review by joshuali925 — DISMISSED** [review]

File: docs/user/dql/functions.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (dai-chen) [reply docs/user/dql/functions.rst:2247]: > > Do you know if the meaning of these params documented in OpenSearch doc or not? I don't see it here: https://opensearch.org/docs/latest/opensearch/query-dsl/full-text/#match-boolean-prefix. I'm thinking if it's missing, probably we can add it in our doc.
> 
> @dai-chen Optional arguments are available in the options section of that page.
> 
> https://opensearch.org/docs/latest/opensearch/query-dsl/full-text/#options

Got it. I missed it. Thanks!

**Comment by joshuali925** [comment]
@MaxKsyunz @forestmvey there are some conflicts after #635 is merged, could you take a look, thanks

File: opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/MatchBoolPrefixQuery.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        .put("minimum_should_match", (b, v) -> b.minimumShouldMatch(v.stringValue()))
+        .put("fuzziness", (b, v) -> b.fuzziness(v.stringValue()))
+        .put("prefix_length", (b, v) -> b.prefixLength(Integer.parseInt(v.stringValue())))
+        .put("max_expansions", (b, v) -> b.maxExpansions(Integer.parseInt(v.stringValue())))
+        .put("fuzzy_transpositions",
+            (b, v) -> b.fuzzyTranspositions(Boolean.parseBoolean(v.stringValue())))
+        .put("fuzzy_rewrite", (b, v) -> b.fuzzyRewrite(v.stringValue()))
+        .put("boost", (b, v) -> b.boost(Float.parseFloat(v.stringValue())))
+        .build());
+  }
```
Comment (Yury-Fridlyand) [inline opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/MatchBoolPrefixQuery.java:32]: `operator` and `analyzer` parameters are missing.
Please add them here and to all tests and docs as well.

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]

**Review by acarbonetto — DISMISSED** [review]

**Review by joshuali925 — APPROVED** [review]

**Review by Yury-Fridlyand — DISMISSED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**